lldb/source/Symbol/SwiftASTContext.cpp

```diff
@@ -2034,39 +2034,49 @@ lldb::TypeSystemSP SwiftASTContext::CreateInstance(lldb::LanguageType language,
 
   // Now if the user fully specified the triple, let that override the one
   // we got from executable's options:
-  if (target.GetArchitecture().IsFullySpecifiedTriple()) {
-    swift_ast_sp->SetTriple(target.GetArchitecture().GetTriple());
-  } else {
-    bool set_triple = false;
-    PlatformSP platform_sp(target.GetPlatform());
+  {
+    auto get_executable_triple = [&]() -> llvm::Triple {
+      if (exe_module_sp)
+        if (auto ts = exe_module_sp->GetTypeSystemForLanguage(
+                lldb::eLanguageTypeSwift))
+          if (auto *exe_ast_ctx = llvm::dyn_cast_or_null<SwiftASTContext>(&*ts))
+            return exe_ast_ctx->GetLanguageOptions().Target;
+      return {};
+    };
+
+    llvm::Triple computed_triple;
     llvm::Triple target_triple = target.GetArchitecture().GetTriple();
-    if (platform_sp && !target_triple.hasEnvironment()) {
-      llvm::VersionTuple version =
-          platform_sp->GetOSVersion(target.GetProcessSP().get());
-      std::string buffer;
-      llvm::raw_string_ostream(buffer)
-          << target_triple.getArchName() << '-' << target_triple.getVendorName()
-          << '-' << llvm::Triple::getOSTypeName(target_triple.getOS())
-          << version.getAsString();
-      swift_ast_sp->SetTriple(llvm::Triple(buffer));
-      set_triple = true;
-    }
 
-    if (!set_triple) {
-      auto type_system_or_err =
-          exe_module_sp->GetTypeSystemForLanguage(lldb::eLanguageTypeSwift);
-      if (!type_system_or_err) {
-        llvm::consumeError(type_system_or_err.takeError());
-        return TypeSystemSP();
+    if (target.GetArchitecture().IsFullySpecifiedTriple()) {
+      computed_triple = target_triple;
+    } else {
+      PlatformSP platform_sp(target.GetPlatform());
+      if (platform_sp && !target_triple.hasEnvironment()) {
+        llvm::VersionTuple version =
+            platform_sp->GetOSVersion(target.GetProcessSP().get());
+        std::string buffer;
+        llvm::raw_string_ostream(buffer)
+            << target_triple.getArchName() << '-'
+            << target_triple.getVendorName() << '-'
+            << llvm::Triple::getOSTypeName(target_triple.getOS())
+            << version.getAsString();
+        computed_triple = llvm::Triple(buffer);
+      } else {
+        computed_triple = get_executable_triple();
```

Comment:
You want to handle the case where `get_executable_triple()` returns `{}`, which I think is equivalent to Dave's point. Otherwise, this LGTM

Reply:
yes this is what I was trying to say, thanks